src/input_stream.rs

```diff
@@ -204,9 +210,19 @@ impl<O: IsA<InputStream> + IsA<glib::Object> + Clone + 'static> InputStreamExtMa
     }
 }
 
+pub struct InputStreamRead<T: InputStreamExtManual>(T);
```

Comment:
Might be useful to also have a function to get back to the original type, and/or implement `Deref` for it

Reply:
Should implement `Eq` and `Debug` here at least, based on the impls of the contained type. Can probably be derived